addons/binding/org.openhab.binding.solarlog/src/main/java/org/openhab/binding/solarlog/handler/SolarLogHandler.java

```diff
@@ -0,0 +1,176 @@
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_YIELDYESTERDAY,
+                SolarLogBindingConstants.CHANNEL_YIELDYESTERDAY, SolarLogBindingConstants.CHANNEL_TYPE_YIELDYESTERDAY));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_YIELDMONTH,
+                SolarLogBindingConstants.CHANNEL_YIELDMONTH, SolarLogBindingConstants.CHANNEL_TYPE_YIELDMONTH));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_YIELDYEAR,
+                SolarLogBindingConstants.CHANNEL_YIELDYEAR, SolarLogBindingConstants.CHANNEL_TYPE_YIELDYEAR));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_YIELDTOTAL,
+                SolarLogBindingConstants.CHANNEL_YIELDTOTAL, SolarLogBindingConstants.CHANNEL_TYPE_YIELDTOTAL));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSPAC,
+                SolarLogBindingConstants.CHANNEL_CONSPAC, SolarLogBindingConstants.CHANNEL_TYPE_CONSPAC));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSYIELDDAY,
+                SolarLogBindingConstants.CHANNEL_CONSYIELDDAY, SolarLogBindingConstants.CHANNEL_TYPE_CONSYIELDDAY));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSYIELDYESTERDAY,
+                SolarLogBindingConstants.CHANNEL_CONSYIELDYESTERDAY,
+                SolarLogBindingConstants.CHANNEL_TYPE_CONSYIELDYESTERDAY));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSYIELDMONTH,
+                SolarLogBindingConstants.CHANNEL_CONSYIELDMONTH, SolarLogBindingConstants.CHANNEL_TYPE_CONSYIELDMONTH));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSYIELDYEAR,
+                SolarLogBindingConstants.CHANNEL_CONSYIELDYEAR, SolarLogBindingConstants.CHANNEL_TYPE_CONSYIELDYEAR));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSYIELDTOTAL,
+                SolarLogBindingConstants.CHANNEL_CONSYIELDTOTAL, SolarLogBindingConstants.CHANNEL_TYPE_CONSYIELDTOTAL));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_TOTALPOWER,
+                SolarLogBindingConstants.CHANNEL_TOTALPOWER, SolarLogBindingConstants.CHANNEL_TYPE_TOTALPOWER));
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Read only
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SolarLogSourceConfig.class);
+        scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    refresh();
+                    updateStatus(ThingStatus.ONLINE);
+                } catch (Exception e) {
```

Comment:
I've implemented something here, not sure if I should go further:

* Data Parsing errors are (currently) handled at a different place 
* JSON Parsing errors are now treated separately
* Not sure the way a HTTP Response error is treated is the best way

I might implement more nuanced handling for IOException (I have not yet understood Jetty's httpClient fully here) but I hope that this is acceptable for the time being.